paz/models/detection/frustum_pointnet.py

```diff
@@ -0,0 +1,177 @@
+from paz.optimization.losses.frustumpointnet_loss import FPointNet_loss
+
+
+def Instance_Seg_Net(point_cloud, one_hot_vec):
+
+    num_points = point_cloud.get_shape().as_list()[1]
+
+    net = tf.expand_dims(point_cloud, 2)
+
+    net = Conv2D(64, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_1')(net)
+
+    net = Conv2D(64, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_2')(net)
+
+    point_feat = Conv2D(64, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_3')(net)
+
+    net = Conv2D(128, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_4')(
+        point_feat)
+
+    net = Conv2D(1024, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_5')(net)
+
+    global_feat = MaxPooling2D(pool_size=[num_points, 1], padding='VALID')(net)
+
+    global_feat = tf.concat([global_feat, tf.expand_dims(tf.expand_dims(one_hot_vec, 1), 1)], axis=3)
+    global_feat_expand = tf.tile(global_feat, [1, num_points, 1, 1])
+    concat_feat = tf.concat(axis=3, values=[point_feat, global_feat_expand])
+
+    net = Conv2D(512, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_6')(
+        concat_feat)
+
+    net = Conv2D(256, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_7')(net)
+
+    net = Conv2D(128, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_8')(net)
+
+    net = Conv2D(128, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_9')(net)
+
+    net = Dropout(rate=0.5)(net)
+
+    logits = Conv2D(2, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation=None, name='conv1_10')(net)
+
+    logits = tf.squeeze(logits, [2])  # BxNxC
+
+    return logits
+
+
+def Box_Est_Net(object_point_cloud, one_hot_vec):
+    num_point = object_point_cloud.get_shape()[1]
+    net = tf.expand_dims(object_point_cloud, 2)
+
+    net = Conv2D(128, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv2_1')(net)
+
+    net = Conv2D(128, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv2_2')(net)
+
+    net = Conv2D(256, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv2_3')(net)
+
+    net = Conv2D(512, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv2_4')(net)
+
+    net = MaxPooling2D(pool_size=[num_point, 1], padding='VALID')(net)
+
+    net = tf.squeeze(net, axis=[1, 2])
+    net = tf.concat([net, one_hot_vec], axis=1)
+
+    net = Dense(units=512, activation='relu')(net)
+    net = Dense(units=256, activation='relu')(net)
+
+    output = Dense(units=3 + NUM_HEADING_BIN * 2 + NUM_SIZE_CLUSTER * 4, activation=None)(net)
+
+    return output
+
+
+def TNet(object_point_cloud, one_hot_vec):
```

Comment:
Add documentation. TNet is also not entirely clear. If this model name is used in the paper describe then it's use in the documentation